torch/csrc/utils/tensor_new.cpp

```diff
@@ -259,6 +277,11 @@ Tensor internal_new_from_data(
     return var.to(device, inferred_scalar_type, /*non_blocking=*/false, /*copy=*/copy_variables);
   }
 
+  bool use_batched_array_extraction = false;
```

Comment:
Why is this block declared here? It's just doing extra work if the data was an array.

Reply:
I pulled out a couple of the parts that were common to https://github.com/pytorch/pytorch/blob/82f0e3ca45d4e52fe5ef782c7c8d318700e328ab/torch/csrc/utils/tensor_new.cpp#L311-L314 and https://github.com/pytorch/pytorch/blob/82f0e3ca45d4e52fe5ef782c7c8d318700e328ab/torch/csrc/utils/tensor_new.cpp#L321-L328 as you asked earlier. If you want I can put the lines ```auto sizes = compute_sizes(data);``` and ```ScalarType inferred_scalar_type = type_inference ? infer_scalar_type(data) : scalar_type;``` back into these two blocks to avoid the extra computation when the data is an array.